File: matlab/shiftT.m
% number of samples to be shifted based on butterworth filter
% di: number of samples to shift
% dt: time to shit
% t: time vector
% y: data vector
% c: pole vector
% n: butterworth filter order
% N: maximum filter order
function [di, dt, tb] = shiftT(t,y,c,n,N)
%{
close all;
t = t10;
y = y10;
n = 6;
N = 10;
%}

tx = linspace(0, length(t)-1, length(t));	% time vector for standard butterworth
[num,den] = genFraq(butterIniC(1,n,N),n);
x = step(num,den,tx);				% generates standard butterworth step response
%[num,den] = genFraq(c,n);	% 

ix5 = 1;	% halfwaypoint butter
ixm = 1;	% peak butter

iy5 = 1;	% halfwaypoint data
iym = 1;	% peak data

while (x(ix5) < 0.5 && x(ix5+1) > 0.5) == false	% finds halfwaypoint of butter
	ix5 = ix5 + 1;
end

[~, ixm] = max(x);		% finds peak of butter
tx = tx/(tx(ixm)-tx(ix5));	% normalizes tx
tx5 = tx(ix5);			% time of halfway point in butter

while (y(iy5) < 0.5 && y(iy5+1) > 0.5) == false	% finds halfway point of data
	iy5 = iy5 + 1;
end


[~, iym] = findpeaks(y(iy5:end));	% finds finds first peak after halfwaypoint in data
if isempty(iym)			% determines if no peaks were found after halfwaypoint
	iym = length(y);	% no peaks found, so last sample is used
else
	iym = iym(1) + iy5;	% peak was found and offset added
end
ty = t/(t(iym) - t(iy5));	% normalizes ty

ty5 = ty(iy5);	% time of halfwaypoint in data

dt = ty5 - tx5;
%di = int32(dt/(ty(1) - ty(2)))
di = -int32(dt/(t(1) - t(2)));
tb = tx;
end

File: MVCFrameWork/src/pro2e/teamX/model/ShiftT.java
//package pro2e.teamX.model;
//
//import pro2e.teamX.matlabfunctions.Matlab;
//import pro2e.teamX.matlabfunctions.MatlabFunktionen;
//import pro2e.teamX.matlabfunctions.SVTools;
//
//public class ShiftT {
//
//	public double[] shiftT(double[] t, double[] y, double[] c, int n, int N) {		
//		
//		int ix5 = 1;
//		int ixm = 1;
//		int ix0 = 1;
//		
//		int iy5 = 1;
//		int iym = 1;
//		int iy0 = 1;
//		
//		int dt, di; 	// Datentyp ???
//		
//		double[] tx;
//		double[] shiftT;
//		double[] genFrag;
//				
//		tx = Matlab.linspace(0, t.length-1, t.length);
//		
//		double genFraq[] = MatlabFunktionen.genFraq(MatlabFunktionen.butterIniC(1, n, N), n);
//		genFraq[num,den] = genFraq(butterIniC(1,n,N),n);
//		
//		
//		double[] x = SVTools.step(B, A, tx);
//		x = SVtool.step(num,den,tx)
//
//		while ((x[ix5] < 0.5 && y[ix5++] > 0.5) == false) {
//			ix5++;
//		}
//		
//		
//		Math.max
//		[~, ixm] = max(x);	% finds peak of butter
//		tx = tx/(tx(ixm)-tx(ix5));	% normalizes tx
//
//		tx5 = tx(ix5);
//		
//		while ((y[iy5] < 0.5 && y[iy5++] > 0.5) == false) {
//			iy5++;
//		}
//		
//		[~, iym] = findpeaks(y(iy5:end));
//		iym = iym(1) + iy5;
//		ty = t/(t(iym) - t(iy5));
//		
//		
//		ty5 = ty * iy5;
//		dt = ty5 - tx5;
//		di = (-1)*(int)(dt/(t[1] - t[2]));
//		
//		shiftT[0] = dt;
//		shiftT[1] = dt;
//		
//		return shiftT;
//	}
//
//}